R/modspec.R

```diff
@@ -39,6 +39,68 @@ write_capture <- function(x) {
   paste0("_capture_[",i-1,"] = ", x[i], ";") 
 }
 
+#' This function adds ETA values to the capture list when the `@etas` 
+#' option is used in `$CAPTURE` .
+#' 
+#' @param x the model object
+#' @param env the mread parse environment
+#' 
+#' @details
+#' We look at env$capture_etas to see if there were any expression text passed
+#' through the `@etas` option. This text will be parsed and evaluated in the 
+#' model environment after adding `LAST` and `last` to represent the last
+#' ETA (or the [total] number of rows in `$OMEGA`).
+#' 
+#' An error is generated in case the expression can't be parsed and evaluated.
+#' 
+#' `@etas` must resolve to an integer-like object. Expecting most usage to be
+#' `1:last` which will be integer, but we want to support `c(1,2,5)` as well
+#' which will not be integer. 
+#'
+#' @return The model object, possibly updated.   
+#' 
+#' @noRd
+capture_etas <- function(x, env) {
+  if(!is.character(env[["capture_etas"]])) return(x)
+  last <- dim_matlist(omat(x))
+  if(last==0) return(x)
```

Comment:
It looks like `dim_matlist` uses `sapply`, so does this `if` condition need to worry about handling a non-scalar `last`?

Reply:
Thanks. This would have been a disaster. I thought I had a method to give the total number of rows for OMEGA but I guess not. For some reason created this `dim_matlist()` function when I already had an `nrow()` method that gives the same result. Thanks again for finding this!

Reply:
Also updated the test to include multiple OMEGA blocks.